src/test/EvolvingProteusProperties.t.sol

```diff
@@ -130,18 +130,27 @@ contract EvolvingProteusProperties is Test {
 
         if (px_final_transformed >= py_final_transformed) {
           vm.expectRevert();
-          DUT = new EvolvingInstrumentedProteus(py_init_transformed, px_init_transformed, py_final_transformed, px_final_transformed, duration);
+          DUT = new EvolvingInstrumentedProteus(py_init_transformed, px_init_transformed, py_final_transformed, px_final_transformed, EVOLUTION_STARTS_IN, duration);
         }
 
         if (px_init_transformed >= py_init_transformed) {
           vm.expectRevert();
-          DUT = new EvolvingInstrumentedProteus(py_init_transformed, px_init_transformed, py_final_transformed, px_final_transformed, duration);
+          DUT = new EvolvingInstrumentedProteus(py_init_transformed, px_init_transformed, py_final_transformed, px_final_transformed, EVOLUTION_STARTS_IN, duration);
         }
     }
 
+    function test_reverts_when_trading_done_before_curve_evolution_starts() public {
+        SpecifiedToken inputToken = SpecifiedToken.X;
+        vm.expectRevert();
+        DUT.swapGivenInputAmount(1e18, 1e18, 1e16, inputToken);
+    }
+
     function testUtilityScaling(int256 x0, int256 y0) public {
         vm.assume(x0 >= int256(MIN_BALANCE) * 2);
         vm.assume(y0 >= int256(MIN_BALANCE) * 2);
+
+        vm.warp(block.timestamp + EVOLUTION_STARTS_IN + 1);
```

Comment:
why do we warp to here instead of EVOLUTION_STARTS_IN + 1. this is going to effectively double the timestamp since EVOLUTION_STARTS_IN = block.timestamp + 1

Reply:
in ForkEvolvingProteus, it warps to EVOLUTION_STARTS_IN

Reply:
because in `testUtilityScalingOverT` I set the new time stamp here ` vm.warp(t_slice);` so eventually it is is `t_slice+ EVOLUTION_STARTS_IN+` as it calls `testUtilityScaling`

Reply:
also `block.timestamp` in foundry unit tests is 0 (the default value)

Reply:
so are you basically just trying to warp to the end of the curve evolution?

Reply:
no in between the evolution duration so when `testUtilityScalingOverT` get's called this is executed 
``` solidity
        t_slice = bound(t_slice, 1, T_DURATION);
        vm.warp(t_slice);
```
which sets the time stamp as the value within the duration and then
`vm.warp(block.timestamp + EVOLUTION_STARTS_IN + 1);` executes where `block.timestamp == t_slice`